varnishgather

```diff
@@ -251,8 +251,18 @@ show_package() {
 	run rpm -qi $1
 }
 
+get_pid () {
+	pidir="$(ls /proc/ | grep -E '^[0-9]+$' 2>&1)"
```

Comment:
I think there are some problems here. First, `pidof` returns parent and children. This this method won't get the varnishd child, which is the more important pid.

2nd, this is a huge breaking change, so what would be best is to first attempt to use `pidof`. If that fails, then try other methods.

Reply:
Also, you are doing a `/proc` scan and status check everytime we need a pid. This should be done once at the top of the gather and the pids stored in variables. It would be best to have 3 variables:

`$PID_VARNISHD` - pid of the parent varnishd process
`$PID_CACHEMAIN` - pid of the cache-main process 
`$PID_ALL_VARNISHD` - both the varnishd and cache-main, space delimited
`$PID_ALL_VARNISHD_COMMA` - both the varnishd and cache-main, comma delimited

Finally, bonus points for this, but the pids we get above need to match the named instance we are after.

Reply:
`get_pid "cache-main"` and `get_pid "varnishd"` returns parent-children respectively for a single varnish instance, but indeed, it doesn't take multiples instances in consideration. I'll re-write the function adding `pidof` as default, if exists, and then try matching the named instance. Thanks for the hints!